src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -61,24 +58,11 @@ template Batch
             let context = show instructionKey
             -- execute instruction and get used holding standards for the instrument
             (instructionCid, instruction) <- fetchByKey @Instruction instructionKey
```

Comment:
Ok I think I understand now. Basically you now rely on the fact that there is a unique "correct" factory implementation taking care of the creation of the correct holding.

I agree it makes sense to remove the sanity check (any other factory implementation will also be vetted, so we can assume their correctness). For this reason, I think it also makes sense to remove the sanity checks on `Credit` and `Debit`, as they are quite expensive to perform.

If we remove those sanity checks, then do we ever need the `getHoldingStandard` utility? 

Reason why I ask is that it's a rather expensive operation (we need from one to three `fetchFromInterface` every time). If we remove the hierarchy we need three `fetchFromInterface` each time.

Reply:
agree, the `Credit` and `Debit` checks are unnecessary, let's rely on the code vetting for improved performance. I removed `getHoldingStandard` and `mustBeSameHoldingStandard` utility functions as they were no longer used.